api/src/main/java/org/apache/iceberg/ManifestFile.java

```diff
@@ -64,14 +64,15 @@ public interface ManifestFile {
       "Summary for each partition");
   Types.NestedField KEY_METADATA = optional(519, "key_metadata", Types.BinaryType.get(),
       "Encryption key metadata blob");
-  // next ID to assign: 520
+  Types.NestedField SCHEMA_ID = optional(520, "schema_id", Types.IntegerType.get(), "Schema ID");
```

Comment:
What is the value of adding the schema ID to a manifest file? Manifests can contain files with different schemas and I don't think there is a way to filter manifests based on the schema of the data files it may contain. At best, you could fill this in if all of the files in the manifest had the same schema ID, but that's really unlikely. I'm not sure it would be worth the code and the metadata.

Reply:
For newly added files, those files should have the same schema ID. I think it should be a common situation that manifest entries have the same schema ID. Such as batch write/overwrite case.

Reply:
Removed the schema ID for manifest file.